### PR TITLE
Kamu 128 implement event details support execute query event

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/dataset-block/metadata-block/components/event-details/components/common/interval-property/interval-property.component.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+@Component({
+    selector: "app-interval-property",
+    templateUrl: "./interval-property.component.html",
+    styleUrls: ["./interval-property.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IntervalPropertyComponent extends BasePropertyComponent {
+    @Input() public data: OffsetInterval | BlockInterval;
+}
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/components/common/interval-property/interval-property.component.ts:16]: Unfortunately, such a trick won't work.
For block intervals we would like to have block links.
For data offset intervals we would finally like to have a redirection to SQL query for this interval.

File: src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts [hunk 2]
```diff
@@ -12,6 +14,10 @@ import {
 })
 export class YamlEventViewerComponent<TEvent extends object> {
     @Input() public event: TEvent;
+    private firstLevelShift = "\u00A0\u00A0";
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts:17]: `readonly`, `static`, UPPERCASE_UNDERSCORE

File: src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts [hunk 3]
```diff
@@ -20,42 +26,66 @@ export class YamlEventViewerComponent<TEvent extends object> {
...
                             item.dataset.kind as string
                         }\n`;
-                        result += `${secondLevelShift}name: ${
+                        result += `${this.secondLevelShift}name: ${
                             item.dataset.name as string
                         }\n`;
-                        result += `${secondLevelShift}owner: ${item.dataset.owner.name}\n`;
+                        result += `${this.secondLevelShift}owner: ${item.dataset.owner.name}\n`;
                     });
+                } else if (key === "inputSlices") {
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts:61]: No need for YAML view in `ExecuteQuery`

File: src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import { SimplePropertyComponent } from "./../common/simple-property/simple-property.component";
+import { EventRowDescriptorsByField } from "../../dynamic-events/dynamic-events.model";
+import { TimePropertyComponent } from "../common/time-property/time-property.component";
+import { IntervalPropertyComponent } from "../common/interval-property/interval-property.component";
+
+export const EXECUTE_QUERY_SOURCE_DESCRIPTORS: EventRowDescriptorsByField = {
+    "ExecuteQuery.DataSlice.logicalHash": {
+        label: "Logical hash:",
+        tooltip: "Logical hash sum of the data in this slice.",
+        presentationComponent: SimplePropertyComponent,
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts:10]: For hashes we need to utilize our hash visualizing component

File: src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        tooltip: "Hash sum of the checkpoint file.",
+        presentationComponent: SimplePropertyComponent,
+        separateRowForValue: false,
+        dataTestId: "execute-query-checkpoint-physical-hash",
+    },
+
+    "ExecuteQuery.Checkpoint.size": {
+        label: "Size:",
+        tooltip: "Size of checkpoint file in bytes.",
+        presentationComponent: SimplePropertyComponent,
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts:58]: For size we need to show our another component that gives bytes/kb/mb

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
Let's also rework `AddData` via dynamic components for complete picture

File: src/app/dataset-block/metadata-block/components/event-details/components/common/time-property/time-property.component.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { ChangeDetectionStrategy, Component, Input } from "@angular/core";
+import { BasePropertyComponent } from "../base-property/base-property.component";
+
+@Component({
+    selector: "app-time-property",
+    templateUrl: "./time-property.component.html",
+    styleUrls: ["./time-property.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TimePropertyComponent extends BasePropertyComponent {
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/components/common/time-property/time-property.component.ts:10]: Can we somehow avoid this wrapper - maybe by unifying the interface of wrapped component?

File: src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts:61]: Removed

File: src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts (hunk 5 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts:58]: Changed on new component with `displaySize` pipe

File: src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts:17]: Fixed

**Comment by zaychenko-sergei** [comment]
The implementation incorrectly draws `ExecuteQuery` event, which has multiple input slices - it repeats each interval in each section:

image

**Comment by zaychenko-sergei** [comment]
It would be great if the identifier of the input dataset was navigable.
Since we have resolved the dataset already to build block links, maybe we can also show it's name and make a hyperlink:

image

**Comment by zaychenko-sergei** [comment]
> The implementation incorrectly draws `ExecuteQuery` event, which has multiple input slices - it repeats each interval in each section:
> 
> image

File: src/app/dataset-block/metadata-block/components/event-details/dynamic-events/builders/event-section.builder.ts [hunk 7]
```diff
@@ -23,7 +30,7 @@ export abstract class EventSectionBuilder<
     public buildEventRows(
         event: TEvent,
         rowDescriptors: EventRowDescriptorsByField,
-        section: keyof TEvent,
+        section: keyof TEvent | keyof ExecuteQueryEventFragment,
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/dynamic-events/builders/event-section.builder.ts:33]: No no no!!!
Why breaking a beautiful abstraction!

Btw, I've noticed we are using resource schema types as `TEvent`, such as `SetPollingSource`, `SetTransform`, while we should probably be using event fragments intsead, such as `SetPollingSourceEventFragment`, `DatasetTransformFragment`

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]

File: src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+    "ExecuteQuery.DataSlice.interval": {
+        label: "Interval:",
+        tooltip: "Data slice produced by the transaction.",
+        presentationComponent: IntervalPropertyComponent,
+        separateRowForValue: false,
+        dataTestId: "execute-query-data-slice-interval",
+    },
+
+    "ExecuteQuery.string.inputCheckpoint": {
```
[the remark this answers is not in the document]
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/components/execute-query-event/execute-query-event.source.ts:33]: > The implementation incorrectly draws `ExecuteQuery` event, which has multiple input slices - it repeats each interval in each section:
> 
> image
> 
> Please also note that in each block 3 of 4 link pairs are incorrect.

File: src/app/components/display-hash/display-hash.component.ts [hunk 9]
```diff
@@ -11,7 +11,7 @@ import { Clipboard } from "@angular/cdk/clipboard";
 })
 export class DisplayHashComponent {
     @Input() public value: string;
-    @Input() public datasetInfo: DatasetInfo;
+    @Input() public datasetInfo?: DatasetInfo;
```
[the remark this answers is not in the document]
Comment (dmitriy-borzenko) [reply src/app/components/display-hash/display-hash.component.ts:14]: Yes, i fixed this input

File: src/app/dataset-block/metadata-block/components/event-details/components/base-dynamic-event/base-dynamic-event.component.ts [hunk 10]
```diff
@@ -26,6 +29,7 @@ export class BaseDynamicEventComponent<TEvent extends object>
     @ViewChildren("container", { read: ViewContainerRef })
     container: QueryList<ViewContainerRef>;
     public isYamlView = false;
+    public isShowYamlToggle = true;
```
[the remark this answers is not in the document]
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/components/base-dynamic-event/base-dynamic-event.component.ts:32]: Moved this property to constructor.

File: src/app/dataset-block/metadata-block/components/event-details/components/common/interval-property/interval-property.component.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+} from "src/app/api/kamu.graphql.interface";
+
+@Component({
+    selector: "app-interval-property",
+    templateUrl: "./interval-property.component.html",
+    styleUrls: ["./interval-property.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IntervalPropertyComponent extends BasePropertyComponent {
+    @Input() public data: OffsetInterval | BlockInterval;
```
[the remark this answers is not in the document]
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/components/common/interval-property/interval-property.component.ts:15]: Fixed

File: src/app/dataset-block/metadata-block/components/event-details/dynamic-events/builders/event-section.builder.ts (hunk 7 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/dynamic-events/builders/event-section.builder.ts:33]: Yes, fixed that. Removed `| keyof ExecuteQueryEventFragment`

**Comment by zaychenko-sergei** [comment]
image

Please try to improve test coverage in the selected files

File: src/app/dataset-block/metadata-block/components/yaml-view-section/yaml-view-section.component.html [hunk 12]
```diff
@@ -0,0 +1,7 @@
+<div
+    class="container-content mt-4"
+    *ngIf="block && (isSetPollingSourceEvent || isSetTransformEvent)"
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/yaml-view-section/yaml-view-section.component.html:3]: There should be a single more general property, such as `isEventWithYamlView`, which internally would be equivalent to current condition.

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]

File: src/app/dataset-block/metadata-block/components/event-details/components/base-dynamic-event/base-dynamic-event.component.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (zaychenko-sergei) [reply src/app/dataset-block/metadata-block/components/event-details/components/base-dynamic-event/base-dynamic-event.component.ts]: Obsolete after recent change in the design

**Comment by zaychenko-sergei** [comment]
Comment from @sergiimk : in `ExecuteQuery` please try to display ID and NAME as separate properties, or even replace ID with NAME

**Comment by zaychenko-sergei** [comment]
Another comment from @sergiimk : in `SetTransform` we cannot use `name` field directly to build a link, instead, it also should be resolved via the identifier property. What is currently called a name is an alias for queries, and it should not be a link, it is just a simple string property - see the schema definition, description field:

```
    "name": {
      "type": "string",
      "format": "dataset-name",
      "description": "An alias of this input to be used in queries."    <<<< This is not obvious, but it could be different from dataset name
    }
```

File: src/app/dataset-block/metadata-block/components/yaml-view-section/yaml-view-section.component.html (hunk 12 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/yaml-view-section/yaml-view-section.component.html:3]: Fixed

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**